### PR TITLE
Template Subfolder convention

### PR DESCRIPTION
**Pull request by kevya-google**
Addresses https://github.com/abcxyz/abc/issues/166 for existing templates in t folder.

Template files to be included are in a folder named "contents" (other options considered were "template", "render", "target", "src", "include")

Updates spec.yaml includes, usage example README, separates READMEs for each template. 
Also adds some default blank values for template inputs to ensure render works according to the READMEs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -185,7 +185,7 @@ https://github.com/hashicorp/go-getter library, including:
 In essence, a template is a directory or directory-like object containing a
 "spec file", named `spec.yaml`
 ([example](https://github.com/abcxyz/abc/blob/main/examples/templates/render/hello_jupiter/spec.yaml)),
-and other files such as source code and config files.
+and other files such as source code and config files. The recommended convention is to have all files intended to be included placed in a subfolder named "contents". This way, non-template related files (GitHub workflows, metadata, template READMEs, etc.) can exist next to the spec.yaml.
```
Comment (drevell) [inline README.md:188]: Let's make this a new paragraph

File: t/react_template/spec.yaml [hunk 2]
```diff
@@ -18,25 +18,34 @@ desc: 'An example template using React library'
 inputs:
   - name: 'automation_service_account'
     desc: 'Automation Service Account (ex: [account_name]@[project_id].iam.gserviceaccount.com)'
+    default: ''
   - name: 'wif_provider'
     desc: 'Workload Identity Federation Provider (ex: projects/[project_id]/locations/global/workloadIdentityPools/[WIF_pool_name]/providers/[provider_name])'
+    default: ''
```
Comment (drevell) [inline t/react_template/spec.yaml:24]: I dunno about this, template rendering will "succeed" but produce broken GH workflows.

**Review by drevell — CHANGES_REQUESTED** [review]

File: t/react_template/spec.yaml (hunk 2 above)
Comment (kevya-google) [reply t/react_template/spec.yaml:24]: When we added the workflows, now inputs for these are required for this template to render.

I think eventually extracting out the workflows/Dockerfiles as discussed in [go/vendor-abcxyz-apps](http://goto.google.com/vendor-abcxyz-apps) is the best move.

But for now, this is blocking the template from being able to be rendered locally, without the workflows. The env file just looks like:

```
AUTOMATION_SERVICE_ACCOUNT=
WIF_PROVIDER=
AR_REPOSITORY=
AR_LOCATION=
CR_SERVICE=
REGION=
PROJECT_ID=
```

with the default blanks. Maybe I can change the defaults to comments of the value hints in the desc for each input?

File: README.md (hunk 1 above)
Comment (kevya-google) [reply README.md]: moved to new paragraph.

**Comment by kevya-google** [comment]
Also, if this is too large to review, I can split this up into smaller PRs, one for updating each template.

File: t/react_template/spec.yaml (hunk 2 above)
Comment (drevell) [reply t/react_template/spec.yaml:24]: I think this is a design problem that we can't just hack around this way. If we're going to include the workflows in the template, we should try hard to make them work. So I think we should either not have defaults or factor them out.

Comment (kevya-google) [reply t/react_template/spec.yaml:24]: Ok, then I think maybe I'll prioritize factoring them out first before continuing with this PR. What do you think?

Comment (drevell) [reply t/react_template/spec.yaml:24]: That sounds fine to me

**Comment by kevya-google** [comment]
Will re-open and split up once .github workflows and Dockerfiles are factored out of existing tempaltes.

**Pull request closed without merging**